bolt-modules/boltlib/lib/puppet/datatypes/resultset.rb

```diff
@@ -17,6 +17,7 @@
       ok_set => Callable[[], ResultSet],
       targets => Callable[[], Array[Target]],
       to_data => Callable[[], Array[Hash]],
+      '[]' => Callable[[Integer], Data]
```

Comment:
I think this has the same return type as `find`, which is `Optional[Variant[Result, ApplyResult]]`

Reply:
I'm having a hard time finding reference material for how to specify an optional parameter with the `Callable` type. Looking at https://puppet.com/docs/puppet/latest/lang_data_abstract.html#the-callable-data-type and https://docs.google.com/document/d/1D1yXXDIw6a67o_hLeqdHWDq2bM41C4eToeJK9tzUc98/edit?usp=sharing but not comming up with the right syntax.

Reply:
Can you use `Variant[Callable[[], Data], Callable[[Integer], Data]]`?

Reply:
ah, yeah. Thanks Nick!